### ISSUE TITLE
Plugin not working with Vagrant 1.8.1 

### ISSUE THREAD

**Issue by jirgn** [mhahn/vagrant-librarian-puppet#47]
When installing your plugin (which worked really well in some of my past projects), I have the following issue on my current setup.
Running 'vagrant up' as well as 'vagrant provision' I get the following Error:

    ==> default: Installing Puppet modules in "puppet" with Librarian-Puppet...
    Unable to load puppet. Please install it using native packages for your platform (eg .deb, .rpm, .dmg, etc).
    puppet --version returned pid 5194 exit 1
    /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/defaults.rb:343: warning: duplicated key at line 344 ignored: :queue_type
    /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/safe_yaml/lib/safe_yaml/syck_node_monkeypatch.rb:42:in `<top (required)>': uninitialized constant Syck (NameError)
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:197:in `require'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:197:in `<module:YAML>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/safe_yaml/lib/safe_yaml.rb:132:in `<top (required)>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/require_vendored.rb:4:in `require'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor/require_vendored.rb:4:in `<top (required)>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor.rb:38:in `require'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor.rb:38:in `require_libs'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/vendor.rb:51:in `load_vendored'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet.rb:164:in `<module:Puppet>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet.rb:29:in `<top (required)>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/util/command_line.rb:12:in `require'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/util/command_line.rb:12:in `<top (required)>'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/bin/puppet:3:in `require'
        from /Users/messner.j/.vagrant.d/gems/gems/puppet-3.4.3/bin/puppet:3:in `<top (required)>'
        from /Users/messner.j/.vagrant.d/gems/bin/puppet:23:in `load'
        from /Users/messner.j/.vagrant.d/gems/bin/puppet:23:in `<main>'

1. **duplicate definition** in ~/.vagrant.d/gems/gems/puppet-3.4.3/lib/puppet/defaults.rb
This can easily be fixed by removing one of the two similar entries. With unistall and reinstall of this plugin the duplicate will be present again.

    :queue_type => {
      :default    => "stomp",
      :desc       => "Which type of queue to use for asynchronous processing.",
    },

2. **uninitialized constant Syck**
Here I did not find any fix, but I'm no ruby developer and am totally confused by the many different versions of ruby and its gems.
Do I have to install the syck gem? And If so - where?

I have the following environment on my System

    $ vagrant --version
    $ Vagrant 1.8.1

    $ vagrant plugin list
    $ vagrant-librarian-puppet (0.9.1)
    $ vagrant-share (1.1.5, system)

    $ ruby --version
    ruby 2.0.0p481 (2014-05-08 revision 45883) [universal.x86_64-darwin14]

**Comment by jirgn** [comment]
downgrading to Vagrant 1.7.4 (using uninstall.tool in vagrant.dmg und reinstall https://releases.hashicorp.com/vagrant/1.7.4/) seems to work for the moment

**Comment by justicel** [comment]
Having the same issue. Seems to be a problem with the new ruby version included in Vagrant 1.8.1. Several other plugins totally broke with the transition.

**Comment by tuxinaut** [comment]
In theory the bug is **simple**. It's a dependency problem between ```puppet``` (version 3.4.3 is currently used) and the ```vagrant``` **ruby**

[Puppet Ruby dependency](http://docs.puppetlabs.com/puppet/3/reference/system_requirements.html#ruby) is **1.9.3**

Vagrant **1.8.1** Ruby version is **2.2.0**

```
{ /opt/vagrant/embedded/bin } -> $ ruby --version
ruby 2.2.0p34 (2015-01-20 revision 49352) [x86_64-linux]
```
It could work if the ```Puppet``` dependency will be updated to a version which supports ```ruby``` >= **2.2.0**

**Comment by tizzo** [comment]
I'm having this issue as well.

**Comment by jhoblitt** [comment]
I have run into this myself.  I believe that puppet 4.x is not officially supported with ruby 2.2 but should generally work.  I am currently testing a version of this plugin with the gemsepec modified to use puppet 4.x and is appears OK thus far.

**Comment by JWGmeligMeyling** [comment]
I'm having this issue as well.

**Comment by jhoblitt** [comment]
This should be resolved by #51

**Comment by jhoblitt** [comment]
0.9.2 has been released!
